ports/stm32/qspi.c

```diff
@@ -204,13 +205,19 @@ STATIC void qspi_write_cmd_data(void *self_in, uint8_t cmd, size_t len, uint32_t
 
     // Wait for write to finish
     while (!(QUADSPI->SR & QUADSPI_SR_TCF)) {
+        if (QUADSPI->SR & QUADSPI_SR_TEF) {
+            ret = -MP_EIO;
```

Comment:
This can just return -MP_EIO, no need to clear TC because it's not set yet (otherwise this loop would exit).  Also TC is cleared on entry to all functions, so it should be OK.